### PR TITLE
make info boxes for posts translatable

### PR DESCRIPTION
**Pull request by andibraeu**
By default there was a line "Posted in <category> on <Month day, Year>" for every language.

This PR enables users to use the translation plugin to translate these strings.

### CODE REVIEW

**Comment by bennothommo** [comment]
@andibraeu Would it be better to use the language strings directly in the component partial, instead of loading them up with the model? You can see an example here: https://github.com/rainlab/user-plugin/issues/402 - which is a similar change to the one you are proposing, but for the User plugin.

**Comment by andibraeu** [comment]
thanks for your feedback.

Unfortunately I didn't get it to work with default values then.

`rainlab.blog::lang.post.posted_on_prefix` will be shown instead of a default value.

**Comment by LukeTowers** [comment]
@andibraeu why do you need default values? What is being passed is a backend language string, so use the filter that's meant to handle those (`| trans`, not `| _`). I.e. `{{ 'rainlab.blog::lang.post.posted_on_prefix' | trans }}`

**Comment by andibraeu** [comment]
thanks for your help, I'm new to twig and octobercms :)

I've made the changes

File: components/Posts.php [hunk 1]
```diff
@@ -230,7 +230,6 @@ protected function listPosts()
                     'slug' => $this->urlProperty($blogPostComponent, 'slug')
                 ]
             );
-
```
Comment (LukeTowers) [inline components/Posts.php]: revert this change please

File: components/post/default.htm [hunk 2]
```diff
@@ -17,11 +17,11 @@
 {% endif %}
 
 <p class="info">
-    Posted
-    {% if post.categories.count %} in
+    {{ 'rainlab.blog::lang.post.posted_on_prefix'|trans }}
```
Comment (LukeTowers) [inline components/post/default.htm]:
```suggestion
    {{ 'rainlab.blog::lang.post.posted_on_prefix' | trans }}
```

File: components/post/default.htm [hunk 3]
```diff
@@ -17,11 +17,11 @@
...
+    {% if post.categories.count %} {{ 'rainlab.blog::lang.post.posted_on_in_category'|trans}}
```
Comment (LukeTowers) [inline components/post/default.htm]: Change the key to `post.posted_in` please

File: components/post/default.htm [hunk 4]
```diff
@@ -17,11 +17,11 @@
...
         {% for category in post.categories %}
             <a href="{{ category.url }}">{{ category.name }}</a>{% if not loop.last %}, {% endif %}
         {% endfor %}
     {% endif %}
-    on {{ post.published_at|date('M d, Y') }}
+    {{ 'rainlab.blog::lang.post.posted_on_mid'|trans }} {{ post.published_at|date('rainlab.blog::lang.post.posted_on_date_format'|trans) }}
```
Comment (LukeTowers) [inline components/post/default.htm]: I would prefer that this was a single translation key where the date was passed in as a translation variable. @bennothommo thoughts on that? Perhaps you could help @andibraeu out if they don't know how to do that.

Comment (andibraeu) [reply components/post/default.htm]: I'm not sure how that should work, as the date format also depends on the translation

Comment (LukeTowers) [reply components/post/default.htm]: you can pass a translated value in as a variable, just starts to get a bit nested. I'm thinking we should do it that way since RTL would have the date before the word

Comment (andibraeu) [reply components/post/default.htm]: I don't have a glue how that should work. Maybe I'm not good enough in reading docs or googling the problem.
But to be honest, I also don't want to get a twig expert, I'm just a user of this plugin that wanted to help getting things improved fast.

Comment (LukeTowers) [reply components/post/default.htm]: Fair enough. @bennothommo when you have time can you clean it up a bit with my suggestions / any of your own ideas?

Comment (bennothommo) [reply components/post/default.htm]: @LukeTowers I agree. I'll try and sort it out tonight.

**Comment by andibraeu** [comment]
any news on that?

**Comment by bennothommo** [comment]
@andibraeu Would you mind testing my changes and let me know if that works for you? Also, feel free to adjust the lang file if necessary - wasn't sure if that's the correct syntax.

@LukeTowers let me know if my changes were what you were after.

**Comment by LukeTowers** [comment]
@bennothommo that looks good to me but can we have spaces around the Twig filter bar? i.e. ` | ` instead of just `|`? In my opinion that makes it much easier to read.

**Comment by andibraeu** [comment]
That change means, that we'll see "in uncategorized" for every post that's not in a catergory?

**Comment by LukeTowers** [comment]
Good catch @andibraeu. @bennothommo I think we should remove the uncategorized fallback

**Comment by LukeTowers** [comment]
Although not sure how to best handle that

**Comment by bennothommo** [comment]
@LukeTowers I did consider that as well, but I didn't think any of the options for that were graceful if we were attempting to simplify the wording down to one language string, ie.

- Have a second language string if no categories are provided.
- Adding an "in " prefix to the `categoryLinks` variable above if categories are present, and remove it from the language string.

I felt having it show up as Uncategorized fits the spirit of the plugin, given that Blog posts are categorised, and matches similar functionality to Wordpress. If people didn't like it, they always had the option of overwriting the partial.

**Comment by LukeTowers** [comment]
@bennothommo I agree with you in principle, however this is a change to the default partial that people are using on their sites out in the wild. You wouldn't expect random strings to start appearing on your frontend after an update so we can't really make that sort of change here.

**Comment by bennothommo** [comment]
@LukeTowers Fair point - I've added a second language string when no categories are present, so the difference in languages can be done purely through the language files. Also, added spaces around the Twig filters.

**Comment by andibraeu** [comment]
are there any news on that?

**Comment by bennothommo** [comment]
@andibraeu The changes are awaiting confirmation that they work before being merged. Have you had a chance to test them?

**Comment by andibraeu** [comment]
yes, I tested it, and I use it. You can see the results here: https://weimarnetz.de/blog

**Comment by bennothommo** [comment]
Thanks @andibraeu!
